parser/object_parse/mod.rs

```diff
@@ -0,0 +1,110 @@
+use std::borrow::{Borrow, Cow};
+
+use fallible_iterator::FallibleIterator;
+use gimli;
+use ir;
+use object::{self, Object};
+use traits;
+use typed_arena::Arena;
+
+use super::Parse;
+
+mod compilation_unit_parse;
+mod die_parse;
+
+use self::compilation_unit_parse::{CompUnitEdgesExtra, CompUnitItemsExtra};
+
+// Helper function used to load a given section of the file.
+fn load_section<'a, 'file, 'input, Sect, Endian>(
+    arena: &'a Arena<Cow<'file, [u8]>>,
+    file: &'file object::File<'input>,
+    endian: Endian,
+) -> Sect
+where
+    Sect: gimli::Section<gimli::EndianSlice<'a, Endian>>,
+    Endian: gimli::Endianity,
+    'file: 'input,
+    'a: 'file,
+{
+    let data = file
+        .section_data_by_name(Sect::section_name())
+        .unwrap_or(Cow::Borrowed(&[]));
+    let data_ref = (*arena.alloc(data)).borrow();
+    Sect::from(gimli::EndianSlice::new(data_ref, endian))
+}
+
+impl<'input> Parse<'input> for object::File<'input> {
+    type ItemsExtra = ();
+
+    fn parse_items(
+        &self,
+        items: &mut ir::ItemsBuilder,
+        _extra: Self::ItemsExtra,
+    ) -> Result<(), traits::Error> {
+        // Identify the file's endianty and create a typed arena to load sections.
+        let arena = Arena::new();
+        let endian = if self.is_little_endian() {
+            gimli::RunTimeEndian::Little
+        } else {
+            gimli::RunTimeEndian::Big
+        };
+
+        // Load the sections of the file containing debugging information.
+        let debug_abbrev: gimli::DebugAbbrev<_> = load_section(&arena, self, endian);
+        let _debug_aranges: gimli::DebugAranges<_> = load_section(&arena, self, endian);
```

Comment:
If we aren't using the section, then I don't think it makes sense to load it. We can just drop this line.